### PR TITLE
Fix: Session test failures (WIP)

### PR DESCRIPTION
**Pull request by seriouslysean**
## Issue number

Relevant issue number: Resolves #

## Please check the following

- [ ] Do the tests still pass? (see [Run the Tests](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#run-the-tests))
- [ ] Is the code formatted properly? (see [Linting (Formatting)](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#linting-formatting))
- For New Features:
  - [ ] Have tests been added to cover any new features or fixes?
  - [ ] Has the documentation been updated accordingly?

### CODE REVIEW

File: config/policies.js [hunk 1]
```diff
@@ -33,44 +37,44 @@ module.exports.policies = {
   },
 
   'game/reconnect': ['isLoggedIn', 'isInGame'],
-  'game/create': ['isLoggedIn', 'hasGameName'],
+  'game/create': [conditionalIsLoggedIn, 'hasGameName'],
```
Comment (seriouslysean) [inline config/policies.js:40]: Could possibly try a different adjustment by adding an opponent specific routes for mock testing that uses the normal route and doesn't have a logged in check.

**Comment by seriouslysean** [comment]
@itsalaidbacklife I think this bug has to do with managing multiple sessions from a singular backend.

https://docs.cypress.io/guides/references/trade-offs

Cypress doesn’t support multiple browsers for our multiple user case, which makes me think we need to mock the cases rather than try and reproduce them in game.

This is why bypassing the auth check in tests only had this pass as far as I can tell. It would be worth going back to verify the tests that failed always had to do with multiple users being registered and actioned on.

Consider our most egregious example, the spectator tests:
- Register the player and log them in
- Express sets the session with the player data
- Register an opponent and log them in
- Express replaces the session data with the opponent 
- Register a spectator and it does it again

The issue then seems to be that the original user isn’t authed properly and gets a 403 when we try and take an action. It would be interesting to try and verify the session ids locally.

That said, there’s a few things I think we can try as ignoring the auth status seems pretty bad, even if it does work. 😬 
- mock non player signups and set them via api calls only, to avoid the socket collisions — we’d still need to blast status to the player somehow
    - This seems like the beat option in terms of standards and stability
- Allow server to manage multiple users in a singular browser session — May work but goes against how the app is actually used
- Add query params to the non player requests to allow them to have slightly different logic (not replace session, etc)
- Bypass auth checks for non player users (will still need a way to not override the express session, maybe some sort of non player middleware in server routes)

**Comment by seriouslysean** [comment]
@Haviles04 🤘🏼 https://github.com/cuttle-cards/cuttle/pull/490

**Comment by itsalaidbacklife** [comment]
Closing for now as this may have been resolved by dependency updates

**Pull request closed without merging**